include/current.h

```diff
@@ -1,67 +1,61 @@
 
+ 
 /*
  * Content: class that measures current using the ACS711EX Hall Effect Sensor
  * Author : Forbes Choy
  * Team   : Sustaingineering UBC
  */
-
+ 
 #pragma once
 #include <Arduino.h>
 #include <farmSensor.h>
-
-
-class current
+ 
+ 
+class current: public farmSensor
 {
 private:
-    int pin; //analog pin number (the GPIO pin number) for data reading
+    int m_pin; //analog pin number (the GPIO pin number) for data reading
     float m_data; //changed to float instead of integer
-    float readRaw(); //returns currentage of source (NOT THE DIVIDER) in decimal value
-
+ 
+protected:
+    float readRaw() override; //returns currentage of source (NOT THE DIVIDER) in decimal value
+ 
 public:
     //constants declarations (cannot declare static const floats in .h files, need .cpp --> 3.9 defined in readRaw())
     static const int adc_res = 4095; //ESP32 ADC resolution is 12 bits
```

Comment:
There is no need to access this variable from outside this class, Let's move to `private` scope